documentation/specs/Pill.md

```diff
@@ -0,0 +1,156 @@
+# `Pill` Component Specification
+
+## Overview
+
+A `Pill` is a compact element that displays contextual text typically representing selections or options; they can be dismissed as well as accompanied by an icon or image.
+
+### Use Cases
+
+- Display selections or options made by the user or system
+
+### Features
+
+- Supports icons and images
+- Supports dismissal
+
+### Risks and Challenges
+
+- Ensuring ease of use
+- Handling invalid prop configurations
+
+### Prior Art
+
+Not a 1 to 1, but similar:
+
+- [MUI `<Chip />`](https://mui.com/material-ui/react-chip/)
+- [Polaris `<Tag />`](https://polaris.shopify.com/components/selection-and-input/tag?example=tag-removable)
+- [React Aria `<TagGroup />`](https://react-spectrum.adobe.com/react-aria/TagGroup.html)
+
+---
+
+## Design
+
+A `Pill` is a fairly straightforward component with no external dependencies and it does not require the use of any React Aria primitives. For icons and dismissal support, this component will follow existing patterns in Easy UI.
+
+### API
+
+```ts
+export type PillProps = {
+  /** Text label */
+  children: ReactNode;
+  /** Left aligned icon */
+  icon?: IconSymbol;
+  /** Left aligned image */
+  image?: ReactNode;
+  /** Callback function when dismissing Pill */
+  onDismiss?: () => void;
+  /** Identifier for Pill */
+  key?: Key;
```

Comment:
**nit**: since `key` is a special prop in React i think we can omit this as a prop that's unique to this component definition